### PR TITLE
[WIP] add minigun

### PR DESCRIPTION
**Pull request by hadim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/minigun/build.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+mkdir -p build
+
+pushd build
+
+cmake -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_PREFIX_PATH=$PREFIX \
+      -DUSE_CUDA=ON \
+      -DBUILD_SAMPLES=OFF \
```
Comment (scopatz) [inline recipes/minigun/build.sh:8]: This should be a Release build

File: recipes/minigun/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{% set name = "minigun" %}
+{% set version = "0.0.1.dev0" %}
```
Comment (scopatz) [inline recipes/minigun/meta.yaml:2]: Is there a fully released version to build from?

File: recipes/minigun/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
```
Comment (scopatz) [inline recipes/minigun/meta.yaml:21]: make should be here too

**Comment by hadim** [comment]
@scopatz thanks for the review but this is WIP (sorry I should have mentioned it). Will ping again when I am done here.

File: recipes/minigun/meta.yaml (hunk 2 above)
Comment (hadim) [reply recipes/minigun/meta.yaml:2]: Ideally, we should wait to have a git tag before merging this. See https://github.com/dglai/minigun/issues/14

**Comment by hadim** [comment]
Not sure this lib really make sense since it is only used in dgl and is headers only. Closing here.

**Pull request closed without merging**